source/server-selection/server-selection.rst

```diff
@@ -849,7 +849,10 @@ For multi-threaded clients, the server selection algorithm is as follows:
 
```

Comment:
adding start and succeeded messages seems reasonable. debug level? they seem equivalent verbosity-wise to command started and command succeeded.

regarding server selection failure, I've generally tried to avoid adding log messages for cases where the user will always see the error via API (and therefore likely in application-generated logs) anyway. I know there are some other "failure" messages e.g. "command failed" and "connection checkout failed", but retryability makes those failures not a 1:1 mapping with driver errors. I think(?) that isn't true of server selection though - are there any cases where server selection can fail but the driver doesn't already throw an error?

Reply:
Do we need a policy on whether to log failures?  In Java driver we took the position that we would not try to catch/log/rethrow exceptions, on the grounds that the application is likely to log exceptions.   This is generally true in Java applications but probably not in other ecosystems.

Reply:
I think it would be good to have a policy on this in the spec. at minimum, I think we should say drivers must not log anything that could (or will definitely) also be reported as an error via API at anything more severe than `debug` level to avoid the possibility of duplicate error-level messages if the app logs them too.

optionally, we could also say that anything which will _definitely_ be reported as an error (like server selection failure) should not be logged at all. for log messages that sometimes (but not always) correspond to an error we throw, it might be overly complicated to ask drivers to only log them if they are not also getting thrown, since the driver likely could not tell that at the place the error condition initially arises. 

I started a thread in driver-devs to ask about application error logging in other language ecosystems. for my languages (copied from the post for posterity): 
* in Swift, per the guidelines [here](https://www.swift.org/server/guides/libraries/log-levels.html) libraries should always prefer to report errors via API and let the application handle logging it.
* in Rust, I’ve seen a bit of a mix of opinions; I see some examples of logging errors but at debug level and also reporting them via API, but also see some people express the opinion that errors should be logged where they are handled, at the app level.